Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponData.cs

```diff
@@ -0,0 +1,36 @@
+namespace Perigon.Weapons
+{
+    public readonly struct WeaponData
+    {
+        public readonly float ElapsedRateOfFire;
+        public readonly float ElapsedReloadDuration;
+        public readonly int ElapsedAmmunitionAmount;
+        public readonly bool IsReloading;
+
+        private WeaponData(
+            float elapsedRateOfFire, 
+            float elapsedReloadDuration, 
+            int elapsedAmmunitionAmount,
+            bool isReloading)
+        {
+            ElapsedRateOfFire = elapsedRateOfFire;
+            ElapsedReloadDuration = elapsedReloadDuration;
+            ElapsedAmmunitionAmount = elapsedAmmunitionAmount;
+            IsReloading = isReloading;
+        }
+
+        public WeaponData Apply(
```

Comment:
Question Why make an `Apply` Method? and not just make the constructor public?

Reply:
Because alot of times you only want to change one value, and keep the rest the same.

And I can apply the variables this.Value

With constructor, there's no this.Value.